lsp/server/src/server.ts

```diff
@@ -129,6 +128,21 @@ connection.onDidChangeConfiguration((change) => {
     connection.languages.diagnostics.refresh();
 });
 
+const MAX_WAIT_FOR_STATE_AGGREGATOR = 4000;
+connection.onDidChangeWatchedFiles((changeEvents) => {
+    changeEvents.changes.forEach((change) => {
+        /**
+        When the default organization changes, the target_id in config.json will be updated. 
+        To handle this file change, we invoke onAuthOrgChanged. 
+        We've noticed that the StateAggregator in the Salesforce code may take over 3 seconds to stabilize, so 
+        we've implemented a fixed delay of up to 4 seconds here.  
+        */
+        if (change.uri.endsWith('.sf/config.json')) {
```

Comment:
What happens if the user is still using `sfdx` instead of `sf`?

Reply:
This is a very good question. I think we can enforce the minimum required core extension version which is inside 'constants.ts'
export const MINIMUM_REQUIRED_VERSION_CORE_EXTENSION = '58.4.1';

Reply:
We can also monitor `config.json' under '.sfdx'

Reply:
Watch  '.sfdx/sfdx-config.json' now. Code is updated.